x-pack/plugins/ingest_manager/common/types/models/epm.ts

```diff
@@ -61,14 +60,24 @@ export interface RegistryPackage {
   description: string;
   type: string;
   categories: string[];
-  requirement: RequirementsByServiceName;
   screenshots?: RegistryImage[];
   icons?: RegistryImage[];
   assets?: string[];
   internal?: boolean;
   format_version: string;
   datasets?: Dataset[];
   config_templates?: RegistryConfigTemplate[];
+}
+
+// Uploaded package archives don't have extra fields
+// Linter complaint disabled because this extra type is meant for better code readability
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface ArchivePackage extends InstallablePackage {}
+
+// Registry packages do have extra fields.
+// cf. type Package struct at https://github.com/elastic/package-registry/blob/master/util/package.go
+export interface RegistryPackage extends InstallablePackage {
+  requirement: RequirementsByServiceName;
```

Comment:
What is this exactly? Trying to find some json examples where this exists.

Reply:
It's used in the UI, I still have a TODO to find out if and how I can recreate that information from a package archive.

Reply:
iirc, this we did (do?) support semver range requirements for Kibana, Elasticsearch, etc.

Will re-read and double check

Reply:
Thanks @jfsiii !

I'll take care of providing that information in a follow-up PR, which will add support for uploaded packages in the UI and all other parts of Fleet that use package assets.